lib/galaxy/auth/providers/ldap_ad.py

```diff
@@ -151,6 +156,15 @@ def ldap_search(self, email, username, options):
                 # setup search
                 attributes = [_.strip().format(**params)
                               for _ in options['search-fields'].split(',')]
+                if string_as_bool(options.get('ldap-autogenerate', False)):
+                    if str(username).lower() == 'none':
+		        attributes.append('displayName')
```

Comment:
You are mixing tabs and spaces, please never use tabs to indent Galaxy Python code, see e.g. https://www.python.org/dev/peps/pep-0008/#indentation

We run the code linter flake8 as part of our CI testing, can you please fix the errors for your pull request?: https://travis-ci.org/galaxyproject/galaxy/jobs/491564860

Reply:
Done. Thank you for yours  detailed explanations.